### PR TITLE
Change language

### PR DESCRIPTION
**Pull request by etiennehomer**
Signed-off-by: HOMER Etienne <etiennehomer@gmail.com>

### CODE REVIEW

File: src/components/TopBar/TopBar.js [hunk 1]
```diff
@@ -236,6 +248,12 @@ const TopBar = ({
         }
     };
 
+    const changeLanguage = (event, value) => {
+        if (onLanguageClick && value !== null) {
```
Comment (ChamseddineBhd) [inline src/components/TopBar/TopBar.js:252]: onLanguageClick !== null 
value couldn't be null.

File: src/components/TopBar/TopBar.js [hunk 2]
```diff
@@ -514,6 +532,68 @@ const TopBar = ({
                                                 </ToggleButtonGroup>
                                             </StyledMenuItem>
                                         )}
+                                        {/*Languages */}
+                                        <StyledMenuItem
+                                            disabled={true}
+                                            style={{
+                                                opacity: '1',
+                                                // padding: '0',
```
Comment (ChamseddineBhd) [inline src/components/TopBar/TopBar.js:540]: remove this commented line

File: demo/src/app.js [hunk 3]
```diff
@@ -128,6 +130,14 @@ const AppContent = () => {
         setEquipmentLabelling(labelling);
     };
 
+    const handleLanguageClick = (currentLanguage) => {
+        setCurrentLanguage(currentLanguage);
+        language =
+            currentLanguage === 'sys'
+                ? navigator.language.split(/[-_]/)[0]
```
Comment (ChamseddineBhd) [inline demo/src/app.js:137]: what if the sys language is Chinese ? should use English if language isn't supported.

**Review by ChamseddineBhd — CHANGES_REQUESTED** [review]

File: src/components/TopBar/TopBar.js (hunk 1 above)
Comment (etiennehomer) [reply src/components/TopBar/TopBar.js:252]: With ToggleButton API, if you click on the button that is already selected, value is null. Thus you can avoid to trigger the function.

File: src/components/TopBar/TopBar.js (hunk 2 above)
Comment (etiennehomer) [reply src/components/TopBar/TopBar.js:540]: Done

File: demo/src/app.js (hunk 3 above)
Comment (etiennehomer) [reply demo/src/app.js]: Yes we said we use english. I made the changes to have this behavior

File: demo/src/app.js [hunk 4]
```diff
@@ -34,14 +34,19 @@ import Button from '@material-ui/core/Button';
...
+    FRENCH,
+} from '../../src/components/TopBar/TopBar';
 
 const messages = {
     en: { ...login_en, ...top_bar_en },
     fr: { ...login_fr, ...top_bar_fr },
 };
 
-const language = navigator.language.split(/[-_]/)[0]; // language without region code
+let language = ENGLISH;
```
Comment (jonenst) [inline demo/src/app.js:49]: make this a state of App

Comment (etiennehomer) [reply demo/src/app.js]: done

File: src/components/TopBar/TopBar.js (hunk 2 above)
Comment (ChamseddineBhd) [reply src/components/TopBar/TopBar.js:539]: Not done yet !

File: src/components/TopBar/TopBar.js [hunk 5]
```diff
@@ -662,6 +741,8 @@ TopBar.propTypes = {
     onAboutClick: PropTypes.func,
     onEquipmentLabellingClick: PropTypes.func,
     equipmentLabelling: PropTypes.bool,
+    onLanguageClick: PropTypes.func,
+    language: PropTypes.string,
```
Comment (ChamseddineBhd) [inline src/components/TopBar/TopBar.js:745]: isRequired ?

**Review by ChamseddineBhd — CHANGES_REQUESTED** [review]

File: src/components/TopBar/TopBar.js (hunk 2 above)
Comment (etiennehomer) [reply src/components/TopBar/TopBar.js]: Done now !

File: src/components/TopBar/TopBar.js (hunk 5 above)
Comment (etiennehomer) [reply src/components/TopBar/TopBar.js]: Ok done

**Review by ChamseddineBhd — APPROVED** [review]

File: src/components/TopBar/TopBar.js [hunk 6]
```diff
@@ -154,6 +159,11 @@ const CustomListItemIcon = withStyles((theme) => ({
 
 export const DARK_THEME = 'Dark';
 export const LIGHT_THEME = 'Light';
+export const SYSTEM = 'sys';
```
Comment (jonenst) [inline src/components/TopBar/TopBar.js:162]: LANG_SYSTEM ?

File: src/components/TopBar/TopBar.js [hunk 7]
```diff
@@ -154,6 +159,11 @@ const CustomListItemIcon = withStyles((theme) => ({
...
+export const ENGLISH = 'en';
+export const FRENCH = 'fr';
+const EN = 'EN';
```
Comment (jonenst) [inline src/components/TopBar/TopBar.js:165]: should be translated in each language, even if the value is the same

Comment (jonenst) [reply src/components/TopBar/TopBar.js:165]: Hmm maybe not.

File: src/components/TopBar/TopBar.js [hunk 8]
```diff
@@ -456,7 +474,6 @@ const TopBar = ({
                                                 disabled={true}
                                                 style={{
                                                     opacity: '1',
-                                                    // padding: '0',
```
Comment (jonenst) [inline src/components/TopBar/TopBar.js:459]: why remove this line

File: src/components/TopBar/TopBar.js [hunk 9]
```diff
@@ -514,6 +531,67 @@ const TopBar = ({
                                                 </ToggleButtonGroup>
                                             </StyledMenuItem>
                                         )}
+                                        {/*Languages */}
+                                        <StyledMenuItem
+                                            disabled={true}
+                                            style={{
+                                                opacity: '1',
```
Comment (jonenst) [inline src/components/TopBar/TopBar.js:538]: why opacity: '1' ?

File: src/components/TopBar/TopBar.js [hunk 10]
```diff
@@ -514,6 +531,67 @@ const TopBar = ({
...
+                                                </Typography>
+                                            </ListItemText>
+                                            <ToggleButtonGroup
+                                                exclusive
+                                                value={language}
+                                                className={
+                                                    classes.toggleButtonGroup
+                                                }
+                                                onChange={changeLanguage}
+                                                aria-label="text formatting"
```
Comment (jonenst) [inline src/components/TopBar/TopBar.js:564]: copypaste

File: src/components/TopBar/TopBar.js (hunk 9 above)
Comment (etiennehomer) [reply src/components/TopBar/TopBar.js:539]: Here StyledMenuItem is disabled as we want to click on the toggle button. But we don't want the item to have disabled style (reduced opacity).

File: src/components/TopBar/TopBar.js (hunk 6 above)
Comment (etiennehomer) [reply src/components/TopBar/TopBar.js]: Done

File: src/components/TopBar/TopBar.js (hunk 8 above)
Comment (etiennehomer) [reply src/components/TopBar/TopBar.js]: It's back

**Review by jonenst — APPROVED** [review]

**Pull request merged**